packages/components/src/SidePanel/SidePanel.component.js

```diff
@@ -39,14 +43,36 @@ function SidePanel({
 	actions,
 	getComponent,
 	components,
-	docked,
+	docked: dockedProp,
 	reverse,
 	minimised,
 	large,
 	dockable,
 	onToggleDock,
 	t,
 }) {
+	const [dockState, setDockState] = useState(dockedProp);
+	const docked = onToggleDock ? dockedProp : dockState;
+	const [width, setWidth] = useState(DEFAULT_MIN_WIDTH);
+	const ref = React.createRef();
+
+	useEffect(() => {
+		if (docked || minimised) {
+			setWidth(large ? LARGE_DOCKED_MIN_WIDTH : DOCKED_MIN_WIDTH);
+		} else if (ref) {
+			const actionList = ref.current.querySelector('.tc-action-list');
```

Comment:
After re-check, `actionList.offsetWidth` will throw an error if we don't have any actions: `.tc-action-list` will not be in the DOM. What do you think?